src/extension.ts

```diff
@@ -41,26 +41,47 @@ export function activate(context: vscode.ExtensionContext) {
 		let isCaseSensitive = false
 		let matchWholeWord = false
 		let filesToInclude = null
+		let filesToExclude = null
+		let useExcludeSettingsAndIgnoreFiles = true
 		let searchText = input
+		let matched = false;
 		for (const it in config) {
 			const data = config[it]
-			isRegex ||= data.isRegex ?? false
-			isCaseSensitive ||= data.isCaseSensitive ?? false
-			matchWholeWord ||= data.matchWholeWord ?? false
-			if (data.filesToInclude)
-				filesToInclude = filesToInclude == null ? data.filesToInclude : filesToInclude + "," + data.filesToInclude
 			let stop = data.stopMatching ?? false
+
 			let key = it.trim()
 			if (key.startsWith("/")) {
 				key = key.substr(1, key.endsWith("/") ? key.length - 2 : key.length - 1)
 				const reg = new RegExp(key, "g")
-				stop = stop && searchText.match(reg) != null
+				matched = searchText.match(reg) != null
+				stop = stop && matched
 				searchText = searchText.replace(reg, data.replacement)
 			}
 			else {
-				stop = stop && searchText.indexOf(key) >= 0
+				matched = searchText.indexOf(key) >= 0
+				stop = stop && matched
 				searchText = searchText.replace(key, data.replacement)
 			}
+
+			if (matched) {
+
+				// regex probably shouldn't revert if a previous non-stopping config sets it to true
+				isRegex ||= data.isRegex ?? false
+
+				// for all other properties, let the matching entry decide these values
+				if (data.hasOwnProperty('isCaseSensitive'))
```

Comment:
js api looks verbose here, can we simplify this condition to `if (data.isCaseSensitive)` ?

Reply:
This is required because if a previous matching entry set the value to true, and then a subsequent matching entry set the value to false, it needs to check for the definition of the value not the truthiness in order to override correctly.

In order to clean it up, we could rely on objects instead of individual variables.  Then it becomes more like so:

```
let finalConfig: SearchConfiguration = {}

for (var it in config) {
   const data = config[it];
   
   // logic to check for match
   
   if (match) {
      finalConfig = {
         ...finalConfig,
         ...match,
      }
   }
}
```

Reply:
in my code with `if (data.isCaseSensitive)` false value will be ignored too. Object sounds good, but I think we can keep separate vars in this PR and rework it later